tests/__init__.py

```diff
@@ -36,7 +35,13 @@ def size(self):
         if self._contents:
             return len(self._contents)
         else:
-            raise RuntimeError('size() is only implemented when contents is provided')
+            try:
+                len(self.contents)
```

Comment:
I'm not sure I understand this change. Whats the difference between self._contents and self.contents?

Reply:
self._contents is the class variable that is optionally initialized when creating the object. self.contents is the class method that actually reads the file (if it's not s3). I'm keeping the original behavior as is and just adding an additional option to read from the file directly if it's not s3.

The reason was I was trying to figure out how to automatically figure out whether batch or inline was expected by just figuring out the size of the file but finding the length of the s3 file seems to be more tricky than expected. I could just check for it being s3 but that's not exactly right. In any case, this update was preparation in case I decide to do more digging into it in the future. Seems fine to just expand functionality but let me know if you disagree.

Reply:
Sounds good. Thanks for the explanation!